front/app/containers/Admin/settings/general/Form.tsx

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import styled from 'styled-components';
+import { object, array, string } from 'yup';
+import { FormProvider, useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { appLocalePairs } from 'containers/App/constants';
+
+// i18n
+import { FormattedMessage, useIntl } from 'utils/cl-intl';
+import messages from '../messages';
+
+// components
+import {
+  IconTooltip,
+  Label,
+  Button,
+  Box,
+} from '@citizenlab/cl2-component-library';
+import {
+  Section,
+  SectionTitle,
+  SubSectionTitle,
+  SectionField,
+  SectionDescription,
+} from 'components/admin/Section';
+import InputMultilocWithLocaleSwitcher from 'components/HookForm/InputMultilocWithLocaleSwitcher';
+import Input from 'components/HookForm/Input';
+import MultipleSelect from 'components/HookForm/MultipleSelect';
+import Feedback from 'components/HookForm/Feedback';
+
+// services
+import { IAppConfigurationSettingsCore } from 'services/appConfiguration';
+
+// Utils
+import { handleHookFormSubmissionError } from 'utils/errorUtils';
+import validateMultilocForEveryLocale from 'utils/yup/validateMultilocForEveryLocale';
+
+const StyledSection = styled(Section)`
+  margin-bottom: 50px;
+`;
+
+interface FormValues {
+  organization_name: IAppConfigurationSettingsCore['organization_name'];
+  locales: IAppConfigurationSettingsCore['locales'];
+  organization_site: IAppConfigurationSettingsCore['organization_site'];
+}
+
+export interface Props {
+  defaultValues: FormValues;
+  onSubmit: (formValues: FormValues) => Promise<void>;
+}
+
+const Form = ({ defaultValues, onSubmit }: Props) => {
```

Comment:
Really love to see one of our most important admin forms getting converted and being so much cleaner 🚀